torch/distributions/multivariate_normal.py

```diff
@@ -131,6 +131,8 @@ def __init__(self, loc, covariance_matrix=None, precision_matrix=None, scale_tri
             if covariance_matrix.dim() < 2:
                 raise ValueError("covariance_matrix must be at least two-dimensional, "
                                  "with optional leading batch dimensions")
+            if not constraints.positive_definite.check(covariance_matrix).all():
+                raise ValueError("covariance matrix must be positive definite")
```

Comment:
Correct. In fact this check is already happening in`super(MultivariateNormal, self).__init__()` below, guarded by `validate_args`.

I think this PR should instead move the `self._unbroadcasted_scale_tril()` calls to be after `super(...).__init__()`, so that arg checking happens before we do any serious linear algebra. WDYT? I think you can repeat the same conditional statements:
```py
super(MultivariateNormal, self).__init__(...)

# Compute unbroadcasted scale_tril after arg validation.
if scale_tril is not None:
   self._unbroadcasted_scale_tril = scale_tril
else:
    self._unbroadcasted_scale_tril = _batch_potrf_lower(covariance_matrix)
```
Note this still misses the linear algebra to compute cov from precision, which might error...
```py
covariance_matrix = _batch_inverse(precision_matrix)
```

Reply:
That's exactly what we did just now 😄 Jeff will push an update in a second.